### PR TITLE
CRM-21392 Allow viewing groups/smartgroups as as other entities (eg. memberships, events)

### PR DESCRIPTION
**Pull request by mattwire**
Overview
----------------------------------------
Add a filter to the "Find Groups" page which changes the "Contacts" link to the selected entity type:

Before
----------------------------------------
You can only view groups from the "Find Groups" screen by Contacts.

After
----------------------------------------
Add a filter to the "Find Groups" page which changes the "Contacts" link to the selected entity type:
localhost_8001_civicrm_group_reset 1

Technical Details
----------------------------------------
The underlying code for groups already supports this and it is already exposed in advanced search criteria.

 * [CRM-21392: Allow viewing groups\/smartgroups as as other entities \(eg. memberships, events\)](https://issues.civicrm.org/jira/browse/CRM-21392)

### CODE REVIEW

**Comment by mlutfy** [comment]
jenkins, test this please

**Comment by mattwire** [comment]
jenkins test this please

**Comment by mattwire** [comment]
test this please

**Comment by mattwire** [comment]
@lcdservices Is this something you would be able to test/review for me?  It's related to #12244 as it exposes the ability to "View As activities/memberships" etc. just like you can do in Advanced search but via the Manage Groups page.

**Comment by lcdservices** [comment]
@mattwire sorry for the delay reviewing this. I implemented and tested and it works as expected. It's a really nice addition to the manage group page!

**Comment by mattwire** [comment]
@eileenmcnaughton Can you tag as "sig:user interfacing improvement"

**Comment by eileenmcnaughton** [comment]
I took a look at the code & it seems largely sensible and contained to the screen @lcdservices tested so merging based on @lcdservices testing

@mattwire ideally we would make this a standard behaviour - the add to group - or was this the missing outlier?

**Comment by mattwire** [comment]
@eileenmcnaughton Can you explain "@mattwire ideally we would make this a standard behaviour - the add to group - or was this the missing outlier?"  I don't quite understand

**Comment by eileenmcnaughton** [comment]
@mattwire I mean - does this now work on all component searches?

File: CRM/Member/Task.php [hunk 1]
```diff
@@ -102,6 +102,16 @@ public static function tasks() {
           'class' => 'CRM_Member_Form_Task_PDFLetter',
           'result' => FALSE,
         ),
+        self::SAVE_SEARCH => array(
```
Comment (eileenmcnaughton) [inline CRM/Member/Task.php]: @mattwire this is the part of the change I'm focussed on

Comment (mattwire) [reply CRM/Member/Task.php]: @eileenmcnaughton Yes, I'm pretty sure it does.  When I added it to CRM/Member/Task I was being cautious but ideally we'd move this into CRM_Core_Task - not sure I've time to thoroughly test that change though at the moment but the groundwork should all be in place for that to be done in the future.

**Pull request merged**